openapi_schema_validator/validators.py

```diff
@@ -63,7 +65,8 @@ def __init__(self, *args, **kwargs):
 
     def iter_errors(self, instance, _schema=None):
         if _schema is None:
-            _schema = self.schema
+            # creates a copy by value from schema to prevent mutation
+            _schema = deepcopy(self.schema)
```

Comment:
I think this could be a shallow copy (`_schema = self.schema.copy()`), at least for the specific issue that's being addressed by this fix. That would be a little more memory efficient than a full (deep) object copy.